// fun-facts/facts.js
module.exports = {
    animals: [
        {
            title: "Kea parrots warble together when they're in a good mood.",
            desc: "This makes them the first known non-mammals to communicate with infectious laughter."
        }
    ],
    body: [
        {
            title: "A baby is born with around 305 bones, but that turns into 206 near adulthood.",
            desc: "Most of a baby's bones are cartilage, but as they grow up most of it turns into bone."
        }
    ],
    brands: [
        {
            title: "Crayola means \"oily chalk.\"",
            desc: "Craie in French means chalk, and ola is short for \"oleaginous,\" or \"oily.\""
        }
    ],
    countries: [
        {
            title: "There's a town called \"llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch.\"",
            desc: "This name supposedly translates to \"St Mary's Church in The Hollow of The White Hazel Near to The Rapid Whirlpool of Llantysilio of The Red Cave.\"",
        }
    ],
    fun: [
        {
            title: "Bananas are curved since they grow towards the sun.",
            desc: "Bananas go through a process called \"negative geotrophism\", so instead of growing towards the ground, they grow towards the sun."
        },
        {
            title: "The inventor of the frisbee was cremated and made into a frisbee when he died.",
            desc: "Walter Morrisson died at 90 in 2010, and his family cremated him, turning the ashes into a frisbee."
        },
        {
            title: "12 + 1 = 11 + 2, just like eleven plus two is an anagram for twelve plus one.",
            desc: "Eleven plus one and twelve plus two also have 13 letters each."
        },
        {
            title: "Central Park's lampposts contain a set of four numbers that can help you navigate.",
            desc: "The first two tell you the nearest street, and the next two tell you whether you're closer to the east or west of the park (even numbers east, odd signals west)."
        },
        {
            title: "Portland was named by a coin flip.",
            desc: "Had the coin landed the other way, the city would be Boston, Oregon."
        },
        {
            title: "In 1997 a Louisville woman left actor Charles Bronson all of her money in a handwritten will.",
            desc: "It totaled to about $300,000. She'd never met him; she was just a fan."
        }
    ],
    language: [
        {
            title: "The official plural of Prius is Prii.",
            desc: "After an online vote, Toyota announced that the official plural of Prius."
        }
    ],
    laws: [
        {
            title: "In switzerland it's illegal to own just one guinea pig.",
            desc: "As guinea pigs are considered as social animals, it is illegal to own less than 2."
        },
        {
            title: "Chewing gum is illegal in Singapore",
            desc: "Gum is illegal to import into Singapore to keep streets clean, though dental or nicotine gun may be exempted."
        }
    ],
    history: [
        {
            title: "King Henry VIII slept with a large axe beside him.",
            desc: "King Henry VIII had a large collection of weapons, which included 6500 handguns."
        },
        {
            title: "During Prohibition, moonshiners would wear \"cow shoes\" to evade police.",
            desc: "The footware left hoofprints instead of footprints to stop the police following their tracks."
        }
    ],
    maths: [
        {
            title: "111,111,111 x 111,111,111 = 12,345,678,987,654,321"
        }
    ],
    sports: [
        {
            title: "In curling, good sportsmanship and politeness are essential.",
            desc: "Congratulating opponents and abstaining from trash talk are part of what\'s known as the \"Spirit of Curling.\""
        },
        {
            title: "The Russians showed up 12 days late to the 1908 Olympics in London.",
            desc: "This happened because they were using the Julian calendar instead of the Gregorian calendar."
        }
    ],
    phobias: [
        {
            title: "There's a fear of feathers: Pteronophobia.",
            desc: "Ptero, in Greek, means \"feather.\""
        }
    ]
}
